### PR TITLE
feat: Create a demo music player page (SSR)

### PR DESCRIPTION
**Pull request by huchshm**
Descriptions
Create a demo music player page 

How to test
/demo  

Link to UI design (if possible):
Provide screenshots or videos that the function is working.
305d9d50ec7c7ce4edb114f5a1c64e7
Provide screenshots of no warning, or error in the console.
dc30b202be648634bd940706f2c978b

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

File: src/pages/demo.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      Demo
+    </audio>
+  );
+};
+
+export default Demo;
+
+export async function getServerSideProps() {
+  const res = await fetch(
+    `https://netease-cloud-music-api-mu-peach.vercel.app/song/url/v1?id=33894312&level=exhigh`
```
Comment (Cameron933) [inline src/pages/demo.tsx:18]: dude, plz hide our API address in the .env file.

File: src/pages/demo.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+interface DemoProps {
+  audioUrl:string;
+}
+
+const Demo = ({ audioUrl }: DemoProps) => {
+  return (
+    <audio role="audio" controls>
```
Comment (Cameron933) [inline src/pages/demo.tsx:7]: This demo do not use any React hooks at all. 
You can improve it with useRef(), a React Hook that lets you reference a value that’s not needed for rendering.

File: src/pages/demo.tsx (hunk 1 above)
Comment (Trent1900) [reply src/pages/demo.tsx:18]: Hi Cam, thank you for the amazing reminder to use process.env to hide the API address, even though that is a server side function, but uploading it to github still expose the sensitive info to the public.
My question is how to get rid of the error about Server Error. see pic below:
31381679302287_ pic_hd

File: src/pages/demo.tsx (hunk 2 above)
Comment (tatoMa) [reply src/pages/demo.tsx:7]: Why do we have to have a hook?

File: src/pages/demo.tsx (hunk 1 above)
Comment (Cameron933) [reply src/pages/demo.tsx]: 1. make sure to put your fetch in an async function.
2. in the .env file, do not write any ; or , after your NEXT_MUSIC_SERVER_ADDRESS variable

**Review by HarryuCATGREY — APPROVED** [review]
LGTM, I use the completed url at your first commit, and it is success to play the music. However, How can I setting my local env for url, I may communicate about it at today's meeting.

**Review by yunshen513 — APPROVED** [review]

**Pull request merged**